recipes/cnpy/all/test_package/conanfile.py

```diff
@@ -0,0 +1,17 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+    def requirements(self):
+        self.requires("gtest/1.10.0")
+    def test(self):
+        if not tools.cross_building(self.settings):
+            bin_path = os.path.join("bin", "test_package")
+            self.run(bin_path, run_environment=True)
```

Comment:
```suggestion

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def test(self):
        if not tools.cross_building(self.settings):
            bin_path = os.path.join("bin", "test_package")
            self.run(bin_path, run_environment=True)
```

Keep it as simple as possible. Don't include new requirements, the idea here is validating the package, not running test packages.